TypeCast assert in SliderTrackMac::owningSliderTrackPart

anil-hebbar opened the pull request:
#### 0208a665a12c58fee1bbc9b47632fbbef5ed6259
<pre>
TypeCast assert in SliderTrackMac::owningSliderTrackPart
<a href="https://bugs.webkit.org/show_bug.cgi?id=289582">https://bugs.webkit.org/show_bug.cgi?id=289582</a>
<a href="https://rdar.apple.com/146390870">rdar://146390870</a>

Reviewed by Ryosuke Niwa and Aditya Keerthi.

The issue was caused by lack of validation in
Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in file. Validation was
added to this and all the other derived classes listed in the radar.

* LayoutTests/ipc/executor.js: Added.
(export.default.Executor):
(export.default.Executor.prototype.createInstance):
(export.default.Executor.prototype.destroy):
* LayoutTests/ipc/remotedisplaylistrecorder-drawcontrolpart-slidertrackpart-crash-expected.txt: Added.
* LayoutTests/ipc/remotedisplaylistrecorder-drawcontrolpart-slidertrackpart-crash.html: Added.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/292274@main">https://commits.webkit.org/292274@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aebd3d59e1c8c22657e1f3df66d9c38a8481df0e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/95295 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/14895 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/4753 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/100340 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/45796 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/15183 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/23326 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/72669 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29942 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/98298 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/11340 "Passed tests") | [![loading-orange](https://github-production-user-asset-6210df.s3.amazonaws.com/3098702/291015173-08c448be-ac0a-4fd6-92a3-8165057445b7.png) 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/86017 "Build is in progress. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; 2 api tests failed or timed out; 2 api tests failed or timed out; Compiled WebKit (warnings); 1 api test failed or timed out; Passed API tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/53002 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/11052 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/3757 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/45135 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/81228 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/3851 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/102377 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/22343 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/16303 "Failed to checkout and rebase branch from PR 42343") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/81667 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/22591 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/82038 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/81067 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20334 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/25636 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/3034 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/15596 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/22313 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/27449 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/21972 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/25446 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/23711 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
```diff
@@ -673,7 +673,7 @@ struct WebCore::IDBDatabaseNameAndVersion {
 
 class WebCore::IDBKeyData {
     bool isPlaceholder()
-    [Validator='WebCore::IDBKeyData::isValidValue(*value)'] std::variant<std::nullptr_t, WebCore::IDBKeyData::Invalid, Vector<WebCore::IDBKeyData>, String, double, WebCore::IDBKeyData::Date, WebCore::ThreadSafeDataBuffer, WebCore::IDBKeyData::Min, WebCore::IDBKeyData::Max> value()
```
Why are we removing this validator check? That doesn't seem right?

anil-hebbar replied:
Not sure - I will take a look at what happened

pxlcoder commented on LayoutTests/ipc/executor.js:
```diff
@@ -0,0 +1,66 @@
+        this._remoteImageBufferSet = this._streamConnection.newInterface('RemoteImageBufferSet', 1234n);
+        // this._coreCrawler = this._streamConnection.newInterface('CoreCrawler', 0n);
+
+        // this._coreCrawler.InitCCDriver();
```
These comments can probably be removed?

anil-hebbar replied:
Updated

pxlcoder reviewed (approved):
LGTM

I think this all lines up with the `isType` implementation in `SPECIALIZE_TYPE_TRAITS_BEGIN` for each ControlPart?

anil-hebbar committed:
TypeCast assert in SliderTrackMac::owningSliderTrackPart
https://bugs.webkit.org/show_bug.cgi?id=289582
rdar://146390870

Reviewed by Ryosuke Niwa and Aditya Keerthi.

The issue was caused by lack of validation in
Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in file. Validation was
added to this and all the other derived classes listed in the radar.

* LayoutTests/ipc/executor.js: Added.
(export.default.Executor):
(export.default.Executor.prototype.createInstance):
(export.default.Executor.prototype.destroy):
* LayoutTests/ipc/remotedisplaylistrecorder-drawcontrolpart-slidertrackpart-crash-expected.txt: Added.
* LayoutTests/ipc/remotedisplaylistrecorder-drawcontrolpart-slidertrackpart-crash.html: Added.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/292274@main
